proxy/grpcproxy/cluster.go

```diff
@@ -48,5 +52,20 @@ func (cp *clusterProxy) MemberUpdate(ctx context.Context, r *pb.MemberUpdateRequ
 
 func (cp *clusterProxy) MemberList(ctx context.Context, r *pb.MemberListRequest) (*pb.MemberListResponse, error) {
 	conn := cp.client.ActiveConnection()
-	return pb.NewClusterClient(conn).MemberList(ctx, r)
+	mresp, err := pb.NewClusterClient(conn).MemberList(ctx, r)
+	if err != nil {
+		return nil, err
+	}
+	if cp.watcher != nil {
+		us, err := cp.watcher.Next()
```

Comment:
Updates are incremental, this probably needs a separate monitoring goroutine that updates a shared slice of members.